Upgrade aws cloudwatch sdk to 4.0.1

vladimir-ilnytskyi opened the pull request:
fix #145 

# Changelog
Upgraded aws cloud watch sdk version to [4.0.1)

vladimir-ilnytskyi committed:
Upgrade aws cloudwatch sdk to 4.0.1

wparad commented:
With this change, I think it would make sense if we could update the version of the library to `4.4` => https://github.com/Cimpress-MCP/serilog-sinks-awscloudwatch/blob/9a5e6daaf1a480df3968c446d977284635f58317/.github/workflows/build.yml#L33

Once that is done, and the build passes, I'll approve and get this merged in.

vladimir-ilnytskyi committed:
Bump build minor version and update gh workflow

vladimir-ilnytskyi commented:
@wparad i have updated a bit workflow file, as I have also introduced .net8 in 
<TargetFrameworks>netstandard2.0;net6.0;net8.0</TargetFrameworks>
So i have updated also build env with 6 and 8 sdks. Not sure how that gonna work but i hope that dotnet commands will publish all 3 runtime versions of nuget

wparad commented:
It won't work. And nuget isn't smart enough to do the right thing. Often what happens is nuget doesn't understand semer and often believes that the latest published version by date is the latest version people should get. Since LTS is on 8, we can dump attempting to build with SDK 6.0 here, and just build and publish with SDK version 8, we don't need to bother that the library can be built with the net6.0 SDK.

wparad commented on .github/workflows/build.yml:
```diff
@@ -2,37 +2,38 @@ on:
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v4
+        with:
+          dotnet-version: |
+            6.0.x
+            8.0.x
```
```suggestion
          dotnet-version: 8.0.408
```

wparad committed:
Fix version to be 8.0.408

wparad commented on .github/workflows/build.yml:
```diff
@@ -2,37 +2,36 @@ on:
+      - name: Run build
+        run: dotnet build
+      - name: Test
+        run: dotnet test -c Release --no-build || exit 1
```
```suggestion
        run: dotnet test -c Release --no-build || exit 1
```
```suggestion
        run: dotnet test -c Release || exit 1
```

wparad committed:
Reverting build skip for test.

vladimir-ilnytskyi commented:
@wparad seems like nuget publish key has been expired.
Could you please check?

error: Response status code does not indicate success: 403 (The specified API key is invalid, has expired, or does not have permission to access the specified package.).

thoean commented:
All good, API key fixed, and v4.4.42 deployed: https://www.nuget.org/packages/Serilog.Sinks.AwsCloudWatch/4.4.42
